src/fancy/Grid.hx

```diff
@@ -190,32 +203,35 @@ class Grid {
     };
   }
 
-  function assignHSize(f: Int -> Float): Int -> Float {
-    if(null != f) return f;
+  function assignHSize(): Int -> Float {
     var cache = new IntCache();
     caches.push(cache);
-    return function(col) {
+
+    return function(col: Int) {
       if(cache.exists(col))
         return cache.get(col);
+
       var v = 0.0;
-      // test headers and first content cell
       var els: Array<Element> = [], el;
-      for(i in 0...(fixedTop + 1).max(2)) {
-        el = renderAt(i, col);
-        els.push(el);
-        view.append(el);
-      }
-      // test last content cell and footer
-      for(i in rows - fixedBottom - 1...rows) {
-        el = renderAt(i, col);
-        els.push(el);
-        view.append(el);
+
+      for (i in 0...rows) {
+        // render each row that is in the fixed top, or the first row after
+        // fixed top, or the last row before fixed bottom, or in fixed bottom
+        if (i < fixedTop + 1 || i > rows - fixedBottom - 1) {
```

Comment:
I know I did, and we should probably switch back, but it was wrong previously for 2x2 grids. If the first row of content was the same as the last row of content, it would try to measure it and remove it from the parent twice. The second time, it would fail, because it had already removed that element from the parent.

This was my quick and dirty solution to at least get it to do the right thing. It's definitely not as efficient, so we should change back to something closer to what you had before.